### PR TITLE
[5.1] Fix tests results on Windows

### PR DESCRIPTION
**Pull request by rentalhost**
Laravel fails in some tests on Windows because it uses double quote as escape character on command-line, Unix uses single quote.

Additionally, it too fixes a `fileinfo` extension dependency on tests. On [Laravel docs](http://laravel.com/docs/5.1) it's not shown as a dependency, so it can be skiped on tests if you don't have enabled this extension without problem.

### CODE REVIEW

File: tests/Console/ConsoleEventSchedulerTest.php [hunk 1]
```diff
@@ -24,20 +27,22 @@ public function testExecCreatesNewCommand()
         $this->assertEquals('path/to/command', $events[0]->command);
         $this->assertEquals('path/to/command -f --foo="bar"', $events[1]->command);
         $this->assertEquals('path/to/command -f', $events[2]->command);
-        $this->assertEquals('path/to/command --foo=\'bar\'', $events[3]->command);
-        $this->assertEquals('path/to/command -f --foo=\'bar\'', $events[4]->command);
-        $this->assertEquals('path/to/command --title=\'A "real" test\'', $events[5]->command);
+        $this->assertEquals('path/to/command --foo='.$escape.'bar'.$escape, $events[3]->command);
+        $this->assertEquals('path/to/command -f --foo='.$escape.'bar'.$escape, $events[4]->command);
```
Comment (GrahamCampbell) [inline tests/Console/ConsoleEventSchedulerTest.php]: please use double quotes an interpolation in all places

Comment (GrahamCampbell) [inline tests/Console/ConsoleEventSchedulerTest.php]: that is, where it makes sense to do so

File: tests/Validation/ValidationValidatorTest.php [hunk 2]
```diff
@@ -1024,6 +1024,9 @@ public function testValidateImage()
         $this->assertTrue($v->passes());
     }
 
+    /**
+     * @requires extension fileinfo
```
Comment (GrahamCampbell) [inline tests/Validation/ValidationValidatorTest.php]: please send this change in a different PR

File: tests/Filesystem/FilesystemTest.php [hunk 3]
```diff
@@ -205,6 +205,9 @@ public function testSizeOutputsSize()
         @unlink(__DIR__.'/foo.txt');
     }
 
+    /**
+     * @requires extension fileinfo
```
Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: same

File: tests/Console/ConsoleEventSchedulerTest.php (hunk 1 above)
Comment (rentalhost) [inline tests/Console/ConsoleEventSchedulerTest.php]: @GrahamCampbell 

It's not possible because Windows accept only double quotes, and Unix uses the `escapeshellarg()`, that will use single quotes. So Symfony reproduced that on `symfony/process` at `ProcessUtils.php:39`.

Seems that we have two PHP bugs related to `escapeshellarg()` on Windows:
- https://bugs.php.net/bug.php?id=43784
- https://bugs.php.net/bug.php?id=49446

An alternative solution is replace single quotes to double quotes on tests, but it'll do exactly the same that I do, but will be even more like a workaround.

Comment (rentalhost) [inline tests/Console/ConsoleEventSchedulerTest.php]: Maybe you mean replaces `'.$escape.'` to `{$escape}`?

I did concat because it's the original style, for instance:
`$this->assertEquals($binary.' artisan queue:listen --tries=3', $events[2]->command);`

Here is used concat instead of interpolation.

But, i'll do that.
